### DISCUSSION TITLE
useMutation response vs MutationCache

### DISCUSSION THREAD

**Discussion by baeharam** [TanStack/query#4970] [Q&A]
I'm using `useMutation` to get `requestId` like below.

```ts
const mutationResponse = useMutation(...);
const requestId = mutationResponse.data?.requestId ?? "";
```
and use `reuqestId` to enable `useQuery`.

```ts
const someQuery = useQuery(..., { enabled: !!requestId });
```

However, when I re-render component, response of `useMutation` is different at every rendering. But, when I log `queryClient.getMutationCache()`, it has `requestId`.

```ts
console.log(mutationResponse.data?.requestId); // undefined or specific request id
console.log(queryClient.getMutationCache()); // always has a specific request id
```

From this point, why mutation response is different at every rendering? Mutation is executed only once at component mounting. It doesn't executed at every rendering. 

Thanks for reading my question! 🙇‍♂️

**Comment by TkDodo** [comment]
> From this point, why mutation response is different at every rendering?

it shouldn't be. can you show a codesandbox reproduction?
